packages/tpetra/core/src/Tpetra_FECrsGraph_def.hpp

```diff
@@ -0,0 +1,335 @@
+ const char tfecfFuncName[] = "FECrsGraph::setup(): ";
+
+ TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC(ownedRowMap.is_null (), std::runtime_error, "ownedRowMap is null.");
+ TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC(ownedPlusSharedRowMap.is_null (), std::runtime_error, "ownedPlusSharedRowMap is null.");
+
+ // If we have a colMap, we're local, otherwise global
+ if(ownedPlusSharedColMap.is_null()) this->allocateIndices(GlobalIndices);
+ else this->allocateIndices(LocalIndices);
+
+ activeCrsGraph_     = Teuchos::rcp(new FEWhichActive(FE_ACTIVE_OWNED_PLUS_SHARED));
+ 
+ // Use a very strong map equivalence check
+ bool maps_are_the_same = ownedRowMap->isSameAs(*ownedPlusSharedRowMap);
+ if(!maps_are_the_same) {
+   // Make an importer if we need to, check map compatability if we don't
+   if(importer_.is_null()) {
+       importer_ = Teuchos::rcp(new import_type(ownedRowMap,ownedPlusSharedRowMap));
+   } 
+   else {
+     TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC(!ownedRowMap->isSameAs(*importer_->getSourceMap()), std::runtime_error, "ownedRowMap does not match importer source map.");
+     TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC(!ownedPlusSharedRowMap->isSameAs(*importer_->getTargetMap()), std::runtime_error, "ownedPlusSharedRowMap does not match importer target map.");
+   }
+
+   // Make sure the ownedPlusSharedRowMap has at least as many entries at the ownedRowMap (due to our superset requriement)
+   TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC( importer_->getNumSameIDs() != importer_->getSourceMap()->getNodeNumElements(),
+                                          std::runtime_error,"ownedRowMap contains entries which are not in the ownedPlusSharedRowMap.");   
+ 
+   TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC( ownedRowMap->getNodeNumElements() > ownedPlusSharedRowMap->getNodeNumElements(),
+                                          std::runtime_error,"ownedRowMap more entries than the ownedPlusSharedRowMap.");   
+
+   // The locallyFitted check is debug mode only since it is more expensive
+   const bool debug = ::Tpetra::Details::Behavior::debug ();
+   if(debug) {
+     TEUCHOS_TEST_FOR_EXCEPTION_CLASS_FUNC( !ownedPlusSharedRowMap->isLocallyFitted(*ownedRowMap),
+                                            std::runtime_error,"ownedPlusSharedRowMap must be locally fitted to the ownedRowMap");
+     
+   }
+
+   // For FECrsGraph, we do all the aliasing AFTER import.  All we need here is a constructor
+   if(ownedPlusSharedColMap.is_null())
+     inactiveCrsGraph_ = Teuchos::rcp(new crs_graph_type(ownedRowMap,0,StaticProfile,params));
+   else
+     inactiveCrsGraph_ = Teuchos::rcp(new crs_graph_type(ownedRowMap,ownedPlusSharedColMap,0,StaticProfile,params));
+ }
+
+}
+
+
+
+template<class LocalOrdinal, class GlobalOrdinal, class Node>
+FECrsGraph<LocalOrdinal, GlobalOrdinal, Node>&
+FECrsGraph<LocalOrdinal, GlobalOrdinal, Node>::
+operator=(const FECrsGraph<LocalOrdinal, GlobalOrdinal, Node>& rhs)
```

Comment:
Wait, I thought you meant for this never to be called.

Reply:
See above